src/pip/_internal/cli/cmdoptions.py

```diff
@@ -79,6 +79,11 @@ def check_dist_restriction(options: Values, check_target: bool = False) -> None:
         options.format_control != binary_only and not options.ignore_dependencies
     )
 
+    if hasattr(options, "dry_run") and options.dry_run:
```

Comment:
```suggestion
    if getattr(options, "dry_run", False):
```

Reply:
But instead of relying on this option name, I wonder if it would be better to use a separate flag that is only set to *True* for `pip install`.

Reply:
TBH I am not sure since I am not that familiar with pip or its architecture. My gut feeling is adding another flag seems too complex. There does not seem to be much reason to differentiate between `install` and `download` regarding dry run. In fact, it sounds to me that there can totally be a dry run mode for `pip download` as well and it would pretty much behave identically with `pip install`.